core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -648,6 +653,7 @@ public PlanOptimizers(
                         SystemSessionProperties::useLegacyWindowFilterPushdown,
                         ImmutableList.of(new WindowFilterPushDown(metadata, typeOperators)),
                         ImmutableSet.of(// should run after DecorrelateUnnest and ImplementLimitWithTies
+                                new RemoveEmptyUnionBranches(),
```

Comment:
Could you explain why the new rule is added in these particular places?

Reply:
This rule can benefit from any action that converts a node into a ValuesNode, so I put them in a few places where that could happen.